src/directory_listing.rs

```diff
@@ -480,40 +483,38 @@ fn sort_file_entries(files: &mut [FileEntry], order_code: u8) -> SortingAttr<'_>
     let mut last_modified = "2";
     let mut size = "4";
 
-    files.sort_by(|a, b| match order_code {
-        // Name (asc, desc)
-        0 => {
-            name = "1";
-            a.name.to_lowercase().cmp(&b.name.to_lowercase())
-        }
-        1 => {
-            name = "0";
-            b.name.to_lowercase().cmp(&a.name.to_lowercase())
-        }
-
-        // Modified (asc, desc)
-        2 => {
-            last_modified = "3";
-            a.modified.cmp(&b.modified)
+    match order_code {
+        0 | 1 => {
+            // Name (asc, desc)
+            files.sort_by_cached_key(|f| f.name.to_lowercase());
```

Comment:
Is there a difference when going with `sort_by_cached_key` here rather than just `sort_by_key` in the scope?
Because what I get from the docs is that `sort_by_cached_key` performs better (cache) for functions with complex computations or operations but can be slower for simple property access, which is the case here.

Reply:
`to_lowercase()` isn’t a simple property access, string manipulation isn’t cheap. I suspect that Rust optimizes for the scenario where the string is already all lowercase however which is the generally case for our test directory (not that we test sorted access there).